sources/PTYSession.m

```diff
@@ -2360,6 +2364,7 @@ - (void)writeTaskNoBroadcast:(NSString *)string
 
 - (void)handleKeyPressInCopyMode:(NSEvent *)event {
     [self.textview setNeedsDisplayOnLine:_copyModeState.coord.y];
+    [self.textview setNeedsDisplayOnColumn:_copyModeState.coord.x];
```

Comment:
This is expensive so only do it if the vertical guide is enabled.

Reply:
Given that this is true any time we call [...textView setNeedsDisplayOnColumn...] I have placed the check there so that most parts of the code don't need to be concerned with this implementation detail (ie the inefficiency of setNeedsDisplayOnCol), but all benefit from this check.